onnxruntime/core/providers/cuda/math/softmax.cc

```diff
@@ -26,25 +26,29 @@ Status SoftMaxComputeHelper(
   auto Y_data = reinterpret_cast<CudaT*>(Y);
   auto X_data = reinterpret_cast<const CudaT*>(X);
 
-  // cudnnSoftmaxForward/Backward is not optimal implementation.
-  // TODO: remove cudnn path completely in the future.
   if (D <= 1024 && D * sizeof(T) <= 4096) {
-    dispatch_softmax_forward<CudaT, CudaT, AccumulationType_t<CudaT>, is_log_softmax>(stream, Y_data, X_data, gsl::narrow_cast<int>(D), gsl::narrow_cast<int>(D), gsl::narrow_cast<int>(N));
-    return Status::OK();
-  }
-
-  std::vector<int64_t> dims({N, 1, 1, D});  // cudnn expects 4D shape in NCHW format
-
-  const auto alpha = Consts<CudaT>::One;
-  const auto beta = Consts<CudaT>::Zero;
-  CudnnTensor input_tensor;
-  CudnnTensor output_tensor;
-  ORT_RETURN_IF_ERROR(input_tensor.Set(dims, CudnnTensor::GetDataType<CudaT>()));
-  ORT_RETURN_IF_ERROR(output_tensor.Set(dims, CudnnTensor::GetDataType<CudaT>()));
-  if (is_log_softmax) {
-    CUDNN_RETURN_IF_ERROR(cudnnSoftmaxForward(handle, CUDNN_SOFTMAX_LOG, CUDNN_SOFTMAX_MODE_INSTANCE, &alpha, input_tensor, X_data, &beta, output_tensor, Y_data));
+    dispatch_warpwise_softmax_forward<CudaT, CudaT, AccumulationType_t<CudaT>, is_log_softmax>(stream, Y_data, X_data, gsl::narrow_cast<int>(D), gsl::narrow_cast<int>(D), gsl::narrow_cast<int>(N));
   } else {
-    CUDNN_RETURN_IF_ERROR(cudnnSoftmaxForward(handle, CUDNN_SOFTMAX_ACCURATE, CUDNN_SOFTMAX_MODE_INSTANCE, &alpha, input_tensor, X_data, &beta, output_tensor, Y_data));
+    const bool use_cudnn_softmax_impl = false;
+    // cudnnSoftmaxForward/Backward is not optimal implementation.
+    // Leave cudnn path for comparision of different input data.
+    if (use_cudnn_softmax_impl) {
```

Comment:
I think we can add a softmax test in *ortmodule_api_test.py, and make sure the numbers matches with various input shape. 
and removed this cudnn path.

Reply:
will add a test. 

Running a benchmark with the WIP benchmarking tool, I find on some data scale, cudnn path runs faster for FP32 mode. 

For example, when input shape is [2400, 12800], original ORT FP32 softmax (using cudnn) is faster than Torch's blockwise implementation on V100. Not sure whether this is reproducible on different models of GPUs. So I would suggest to leave CUDNN branch there still, which could be an alternative when we trying to tuning the kernels according to data scales. What do you think about this? 

![image](https://user-images.githubusercontent.com/10530022/141385971-c95b8afa-937a-4787-9dfe-0e52ce948f69.png)

Reply:
talked offline, let's remove the cudnn path now since fp32 "regression" is not a concern for us now.